read.go

```diff
@@ -269,10 +266,52 @@ func readNativeFrames(d dicomio.Reader, parsedData *Dataset, fc chan<- *frame.Fr
 
 	debug.Logf("readNativeFrames:\nRows: %d\nCols:%d\nFrames::%d\nBitsAlloc:%d\nSamplesPerPixel:%d", MustGetInts(rows.Value)[0], MustGetInts(cols.Value)[0], nFrames, bitsAllocated, samplesPerPixel)
 
+	bytesAllocated := bitsAllocated / 8
+	bytesRead = bytesAllocated * samplesPerPixel * pixelsPerFrame * nFrames
+	if bitsAllocated == 1 {
+		bytesRead = pixelsPerFrame * samplesPerPixel / 8 * nFrames
+	}
+
+	skipOne := false
+	switch {
+	case uint32(bytesRead) == vl: // we good nothing
+	case uint32(bytesRead) == vl-1 && vl%2 == 0:
+		skipOne = true
+	case uint32(bytesRead) == vl-1 && vl%2 != 0:
+		return nil, 0, fmt.Errorf("vl=%d: %w", vl, ErrorExpectedEvenLength)
+	default:
+		if !parsedData.opts.allowMismatchPixelDataLength {
+			return nil, 0, fmt.Errorf("expected_vl=%d actual_vl=%d %w", bytesRead, vl, ErrorMismatchPixelDataLength)
+		}
+		data := make([]byte, vl)
+		_, err = io.ReadFull(d, data)
+		if err != nil {
+			panic(err)
+		}
+
+		f := frame.Frame{
+			Encapsulated: true,
+			EncapsulatedData: frame.EncapsulatedFrame{
+				Data: data,
+			},
+		}
```

Comment:
I'm not sure I like this model of parsing is as encapsulated Frame when it isn't actually supposed to be an encapsulated frame. Like trying to use this data as an encapsulated frame wouldn't work as expected later. Maybe we need some kind of error flag to indicate this was a fallback in the frame element.

Can you tell me more about how you need to access this data in your use case? Like if you hit a dicom with this error condition, do you just want to skip doing any downstream work with the pixel data? Or mark the pixel data as unusable or something? Because iiuc in the current model you have no way of distinguishing this case from a case where there was just a valid encapsulated frame parsed (if your option is set to true).

Reply:
I added another variable at dd6ececbb145ef4c4a4d6baeb9042300823dbd9d (IsVLMismatch).

- Without `AllowMismatchPixelDataLength`, if the VL mismatches calculated `BytesToRead`, the library returns an error.
- With `AllowMismatchPixelDataLength`, the library return a Frame with `VLMismatch=true` indicates that the pixel data is unusable.